proto/astarteplatform/msghub/property.proto

```diff
@@ -0,0 +1,64 @@
+/*
+ * This file is part of Astarte.
+ *
+ * Copyright 2024 SECO Mind Srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+syntax = "proto3";
+
+package astarteplatform.msghub;
+
+import "astarteplatform/msghub/astarte_type.proto";
+
+/* Enum representing an Astarte interface ownership. */
+enum Ownership {
+  DEVICE = 0;
+  SERVER = 1;
+}
+
+/* A message representing the property value associated to a certain interface and path. */
+message Property {
+  string path = 1;                      // Property path.
+  AstarteDataTypeIndividual value = 2;  // Astarte data.
+}
+
+/* A message containing all the properties values and information associated to a given astarte interface. */
+message InterfaceProperties {
+  Ownership ownership = 1;            // Interface ownership.
+  uint32 version_major = 2;           // Interface major version.
+  repeated Property properties = 3;   // A list of Properties values.
+}
+
+/* This message is the response to the GetProperties rpc. */
+message GetPropertiesResponse {
+  map<string, InterfaceProperties> interface_properties = 1; // Map interface names to interface property data.
+}
+
+/* This message is the request to the GetAllProperties rpc.
+ * If the ownership is not specified, all the interfaces are retrieved, both device and server owned.
+ * Otherwise, only the interfaces with the specified interfaces are retrieved.
+ */
+message GetAllPropertiesRequest {
+  optional Ownership ownership = 1;   // Optional field representing the ownership of the interface to retrieve.
+}
+
+/* This message is the response to the GetProperty rpc. */
+message GetPropertyRequest {
```

Comment:
just `Property`?

Reply:
I've renamed the request/response messages, let me know what you think